sumologic/resource_sumologic_metadata_source.go

```diff
@@ -0,0 +1,279 @@
+package sumologic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceSumologicMetadataSource() *schema.Resource {
+	pollingMetadataSource := resourceSumologicSource()
+	pollingMetadataSource.Create = resourceSumologicMetadataSourceCreate
+	pollingMetadataSource.Read = resourceSumologicMetadataSourceRead
+	pollingMetadataSource.Update = resourceSumologicMetadataSourceUpdate
+	pollingMetadataSource.Importer = &schema.ResourceImporter{
+		State: resourceSumologicSourceImport,
+	}
+
+	pollingMetadataSource.Schema["content_type"] = &schema.Schema{
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.StringInSlice([]string{"AwsMetadata"}, false),
+	}
+	pollingMetadataSource.Schema["scan_interval"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Required: true,
+	}
+	pollingMetadataSource.Schema["paused"] = &schema.Schema{
+		Type:     schema.TypeBool,
+		Required: true,
+	}
+	pollingMetadataSource.Schema["url"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Computed: true,
+	}
+	pollingMetadataSource.Schema["authentication"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		ForceNew: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"type": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringInSlice([]string{"S3BucketAuthentication", "AWSRoleBasedAuthentication"}, false),
+				},
+				"access_key": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"secret_key": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"role_arn": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+		},
+	}
+	pollingMetadataSource.Schema["path"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		ForceNew: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"type": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringInSlice([]string{"S3BucketPathExpression", "CloudWatchPath", "AwsMetadataPath"}, false),
+				},
+				"bucket_name": {
```

Comment:
bucket name and path expression are not needed by the AWS metadata source, why do we add them in the schema?

Reply:
good catch! will fix that.